amundsen_application/models/user.py

```diff
@@ -86,11 +86,20 @@ def make_user(self, data: Dict) -> User:
 
     @validates_schema
     def validate_user(self, data: Dict) -> None:
-        if not data.get('display_name', None):
-            raise ValidationError('"display_name" must be provided')
+        if _str_no_value(data.get('display_name')):
+            raise ValidationError('"display_name", "full_name", or "email" must be provided')
 
-        if not data.get('user_id', None):
-            raise ValidationError('"user_id" must be provided')
+        if _str_no_value(data.get('user_id')):
+            raise ValidationError('"user_id" or "email" must be provided')
+
+
+def _str_no_value(s: Optional[str]) -> bool:
```

Comment:
not sure if you need a function for this. Would `not s` serve the same purpose?

Reply:
I see, so there is a case like s = ' '?

Reply:
Yes, we were getting and empty string for the `display_name`, and the previous code made it such that it accepted the empty string instead of trying to assign `full_name` or `email` as a fallback.

Reply:
I was playing with the python interpreter and `not ''` returns `True`. I'm curious why the if block on line 78 does not run.
```if not data.get('display_name', None):```

Reply:
it should run, my understanding is that the display_name could be empty string with a space? In that case it will be true when evaluating.

Reply:
I don't recall ever seeing any responses like `" "` returned from metadata service. The fields are mostly `""` when empty. Not sure if this is the actual root cause of the bug.

Reply:
The `"  "` is not the root cause of the bug, but accounting for it is just an extra precaution. The only contract between frontend and metadata is that this value is going to be a string.